### ISSUE TITLE
Changing repeat cron timer won't be updated in redis

### ISSUE THREAD

**Issue by weeco** [OptimalBits/bull#752]
Apparently it's not possible to update/edit the cron timers without manually flushing the database or removing the repeating job. They will always remain the same. Below a code snippet to reproduce the issue on 3.3.0:

```
const testQueue = new Queue('tests', { redis: config.redisBull })
testQueue .add('buggy job', null, { repeat: { cron: '*/2 * * * *' }, removeOnComplete: true, removeOnFail: true })
testQueue .process('buggy job', lengthyJob)

function lengthyJob(_job) {
  logger.info('lengthy job started')
  return Promise.delay(30 * 1000).then(() => {
    logger.info('Job is done')
    return 'done'
  })
}
```

This will run the lengthy job every 2 minutes. If you apply * * * * * as cron in a second run it will still be executed only every 2 minutes.

**What I have expected Bull to do instead:**
I have expected it checking for existing repeating jobs with that name. If one with that name already exists, it would simply always overwrite the cron when it gets added.

**Comment by manast** [comment]
yes, I am leaning to the solution where you can only have one cron job per name. The only problem is that it is a breaking change that would require a major release, unless we consider the current behaviour as a bug.

**Comment by weeco** [comment]
In what case it would be a breaking change? If I change the cron timer it won't update the existing repeating job, nor would it create a new one. Unless this bug is considered a feature, it woudn't be a breaking change I guess  :D

**Comment by manast** [comment]
In the sense that you can have several cron jobs under the same name, instead of only one cron job per name.

**Comment by weeco** [comment]
Actually I wonder how this would work, since after the edit of the cron timestamp it wouldn't add a second job with a different cron timer. I am not sure if this feature works at all right now?

**Comment by ritter** [comment]
My opinion is to have repeat jobs have a unique name, and not being unique by cron value. But it all broke for me in v3.3.2 and I'm having to keep back at v3.3.1 for repeat jobs.

**Comment by Freundschaft** [comment]
I also think that having multilpe repeat jobs per name is kind of useful, since named processors work better with that kind of schema. Otherwise Id have to create a named processor per cronjob?

**Comment by manast** [comment]
@Freundschaft yes, you will need to specify one named processor per cronjob, but you could specify the same one several times :).

**Comment by Freundschaft** [comment]
ah, makes sense thanks

**Comment by Freundschaft** [comment]
ok I just figured out that a repeat job always gets a custom jobId like this

> repeat:9f83cc13f62b7ea58d6c9684a06eab1b:1510773300000

no matter what I set in the jobId option, it gets overwritten. can this behavior be changed?

**Comment by manast** [comment]
that is an internal key representation of the job, why would you need to "parse" it?

**Comment by Freundschaft** [comment]
I was referring to the solution outlined by @ritter and thought giving a unique id to the job by overriding the jobid could be a solution. It seems this is not possible for repeat jobs if I am not mistaken, right?

**Comment by manast** [comment]
@Freundschaft I do not know which solution you are referring to, can you copy it here or put a link?

**Comment by Freundschaft** [comment]
its just a theoretical solution.
The idea is to give every repeat job a unique name, so we can update it by its unique name later on.
e.g. I have 10 shops that need synchronization every hour, so I'd like to add 10 repeat jobs, each having the unique job id of the shop id that the job belongs to.
that way I can update the cronjobs or ensure that they are running, on application start, without duplicating them.
